QuackNBOOM/QuackNBOOM/Bomb.swift

```diff
@@ -0,0 +1,109 @@
+//
+//  Bomb.swift
+//  QuackNBOOM
+//
+//  Created by Ngo Tuyetnhi N. and Benoit Neriah R. on 3/4/18.
+//  Copyright © 2018 Neriah and Neenee. All rights reserved.
+//
+// Bomb is derived from GameObject. It's weapon used by the player to kill the ducks.
+//      The bomb goes to the tapped area and explodes after a countdown.
+//      The explosion works by hiding the bomb is displaying the explosion - two different sprites
+
+import Foundation
+import SpriteKit
+
+class Bomb: GameObject {
+    let explosionSprite: SKSpriteNode           // Another sprite for the explosion
+    let startExplodeCountdown = CGFloat(0.5)    // The value as to what to reset the countdown to
+    var explodeCountdown = CGFloat(0.5)         // The explosionCountdown explodes the bomb when it reaches 0. When the bomb has been exploded, it's set to -100
+    
+    
+    
+    // Initializes both sprites and the attributes
+    //
+    // - Parameters:
+    //      - imagePath: bomb.sprite image
+    //      - explosionImagePath: explosionSprite image
+    required init(imagePath: String, explosionImagePath: String){
+        // Set images
+        explosionSprite = SKSpriteNode(imageNamed: explosionImagePath)
+        super.init(imagePath: imagePath)
+        
+        // Set attributes
+        zPos = 2.0
+        scale = CGSize(width: 0.75, height: 0.75)
+        
+        // Set the sprites' visibility
+        sprite.isHidden = false
+        explosionSprite.isHidden = true
+    }
+    
+    // Initializer. Automatically sets the explosion sprite
+    //
+    // - Parameter: imagePath: bomb.sprite image
+    required init(imagePath: String){
+        // Set images
+        explosionSprite = SKSpriteNode(imageNamed: "Explode")   // Automatically sets image name
+        super.init(imagePath: imagePath)
+        
+        // Set attributes
+        zPos = 2.0
+        scale = CGSize(width: 0.75, height: 0.75)
+        
+        // Sets the sprites' visibility
+        sprite.isHidden = false
+        explosionSprite.isHidden = true
+    }
+    
+    
+    // Resets the explode countdown and the sprites' visibility
+    //
+    func resetExplosionCountdown() {
+        explodeCountdown = startExplodeCountdown
+        
+        // Sets the sprites' visibility
+        sprite.isHidden = false
+        explosionSprite.isHidden = true
+    }
+    
+    // Update - runs  once every frame. Sets the sprites attributes to those of this Bomb
+    //
+    // - Parameter deltaTime: the amount of time between each frame
+    override func update(_deltaTime: TimeInterval) {
+        super.update(_deltaTime: _deltaTime)
+        // Update the explosionSprite's attributes
```

Comment:
If you add the explosionSprite to the bomb's sprite (aka self) it should track the position automatically, since its parent is moving and probably only need to do it in the init.
```
super.init(imagePath: imagePath)
addChild(explosionSprite)
```

Reply:
An *NSexception* occurs in the *APPDelegate* class. The error says; *"Thread 1: signal SIGABRT"*.

![Screenshot of Error](https://i.imgur.com/fX3C8cK.png)


Attempted with the following:
```swift
super.init(imagePath: imagePath)
addChild(explosionSprite)
```
```swift
super.init(imagePath: imagePath)
explosionSprite.removeFromParent()
addChild(explosionSprite)
```

Reply:
Hmm, should work. Show me in class and we'll get it working. Its worth noting that you must of implemented
```
class GameObject: SKSpriteNode {...
```
before you can addChild(:). Something like this:
```
     required init(image: String, explosionImage: String){
         // Set images
         explosionSprite = SKSpriteNode(imageNamed: explosionImage)
         super.init(imagePath: image)
         addChild(explosionSprite)
```

Reply:
Alright, I’ll show you in class this week.
Just to confirm; yes, *GameObject* inherited from *SKSpriteNode* at the time of the attempt.